superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -218,14 +226,7 @@ export default class ResultSet extends React.PureComponent {
         </Alert>
       );
     } else if (query.state === 'success' && query.ctas) {
-      // Async queries
-      let tmpSchema = query.tempSchema;
-      let tmpTable = query.tempTableName;
-      // Sync queries, query.results.query contains the source of truth for them.
-      if (query.results && query.results.query) {
-        tmpTable = query.results.query.tempTable;
-        tmpSchema = query.results.query.tempSchema;
-      }
```

Comment:
this logic is moved to the reducer, it seems cleaner (although it's a big janky to have the query in 2 different places here)